test/goalc/test_arithmetic.cpp

```diff
@@ -0,0 +1,232 @@
+// https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests
+
+#include <thread>
+#include <chrono>
+
+#include "gtest/gtest.h"
+#include "game/runtime.h"
+#include "goalc/listener/Listener.h"
+#include "goalc/compiler/Compiler.h"
+
+#include "third-party/inja.hpp"
+#include "third-party/json.hpp"
+#include <common\util\FileUtil.h>
+
+#include <test\goalc\framework\test_runner.h>
+
+#include <iostream>
+#include <string>
+#include <cstdio>
+#include <sstream>
+#include <iostream>
+#include <random>
+
+// --------
+// This is a very over-engineered integer test, but it serves as a decent example of how to use the
+// test+template framework
+// I've heavily annotated it with comments and links to docs to help
+// --------
+
+// We are using Google Test's paramaterized test feature
+// This allows us to define a single generic test, and pass in a whole bunch of values
+// See - https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#value-parameterized-tests
+struct IntegerParam {
+	// An index is needed to be explicitly set because I couldn't find a way to pull the test-index number from google's API
+	// TODO - if you can find a way, please improve!
+	// But this is needed so we can uniquely save the template files, especially if they error out
+	// Why? - since you may choose to generate random values, it's nice for them to be stored after the tests complete.  Some tests may be complex as well
+  int index;
+	// Each integer test has a signed value, and can be represented as hex or an integral
+  s64 val;
+  bool hex;
+
+  IntegerParam(s64 val, bool hex = false, int index = 0) : val(val), hex(hex), index(index) {}
+
+	// This is used to generate the value that is passed into the template engine
+	// and injected into the file of lisp code.
+	// In most cases this will probably be a string but look into inja's capabilities
+	// - https://github.com/pantor/inja
+  std::string toLisp() {
+    // Append hex reader macro '#x'
+    if (hex) {
+      return std::string("#x") + std::string(std::to_string(val));
+    }
+    return std::to_string(val);
+  }
+
+	// This is used by the test runner code to know what the expected value is
+	// For a simple example like this, a single eval is all that's required, but for
+	// more complex tests, this may not be the case.
+  std::string eval() {
```

Comment:
This is great and will probably be super useful for more complicated randomly generated expressions.